### PR TITLE
Tidy long run artifacts

### LINKED ISSUE #730: Remaining to do for Diagnostics
- [x] Comprehensive tests #743 
- [x] Finalize docs #743 
- [x] Make sure carbon flux units are consistent, make a choice (kg or mol, CO2 or C) #731 
- [x] Runoff model: runoff variables are present or not depending on RunoffModel choice
- [x] Possibly, in define_diagnostics comments, add "2D" / "3D" or "surface only" / "subsurface" #734 
- [x] Finalize use of Diagnostics in all experiments and tests and docs tutorials (refactoring)
- [x] Bucket/EnergyHydrology do not define output_type and frequency #734 
- [ ] Make buildkite artifacts more navigable #874 
- [ ] Add one small integration test in `test` with a couple of diagnostics #874

### PR DESCRIPTION
**Pull request by AlexisRenchon**
- [x] one .pdf instead of 36 .png artifacts in long runs
- [x] add diagnostic tests in CI pipeline

closes #730

### CODE REVIEW

File: test/diagnostics/diagnostics_tests.jl [hunk 1]
```diff
@@ -4,7 +4,7 @@ using ClimaLand.Diagnostics: @with_error
 
 @test isdefined(ClimaLand.Diagnostics, :compute_sw_albedo!)
 
-@test !hasmethod(
+@test hasmethod(
```
Comment (Sbozzolo) [inline test/diagnostics/diagnostics_tests.jl:7]: I think this should be `!`. The method gets defined below by `@diagnostic_compute`

File: .buildkite/Manifest-v1.11.toml [hunk 2]
```diff
@@ -2193,6 +2205,12 @@ git-tree-sha1 = "36d8b4b899628fb92c2749eb488d884a926614d3"
 uuid = "2dfb63ee-cc39-5dd5-95bd-886bf059d720"
 version = "1.4.3"
 
+[[deps.Poppler_jll]]
+deps = ["Artifacts", "Cairo_jll", "Fontconfig_jll", "Glib_jll", "JLLWrappers", "JpegTurbo_jll", "Libdl", "Libtiff_jll", "OpenJpeg_jll", "Pkg", "libpng_jll"]
+git-tree-sha1 = "02148a0cb2532f22c0589ceb75c110e168fb3d1f"
+uuid = "9c32591e-4766-534b-9725-b71a8799265b"
+version = "21.9.0+0"
```
Comment (Sbozzolo) [inline .buildkite/Manifest-v1.11.toml:2212]: There should be a more recent version of Poppler available (not that it matters too much)

File: test/diagnostics/diagnostics_tests.jl (hunk 1 above)
Comment (AlexisRenchon) [reply test/diagnostics/diagnostics_tests.jl:7]: The test fails with!.
Let's understand what is happening

File: experiments/long_runs/soil.jl [hunk 3]
```diff
@@ -365,6 +367,7 @@ setup_and_solve_problem(; greet = true);
 #### ClimaAnalysis ####
 simdir = ClimaAnalysis.SimDir(outdir)
 short_names = ["swc", "si", "sie"]
+tmpdir = mktempdir(root_path)
```
Comment (Sbozzolo) [inline experiments/long_runs/soil.jl:370]: Maybe let's be more explicit in cleaning up the folder and do 
```suggestion
mktempdir(root_path) do tmpdir
```
(and add an `end`) at the end of this file

**Review by Sbozzolo — APPROVED** [review]

**Comment by Sbozzolo** [comment]
Benchmarks fail for unrelated reasons. I think this is ready to be merged.

**Pull request merged**